aws/resource_aws_iam_service_linked_role_test.go

```diff
@@ -0,0 +1,90 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsIamServiceLinkedRole(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsIamServiceLinkedRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccAwsIamServiceLinkedRoleConfig, "elasticbeanstalk.amazonaws.com"),
+				Check:  testAccCheckAwsIamServiceLinkedRoleExists("aws_iam_service_linked_role.test"),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsIamServiceLinkedRoleDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).iamconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_iam_service_linked_role" {
+			continue
+		}
+
+		service_name := rs.Primary.ID
+
+		params := &iam.ListRolesInput{
+			PathPrefix: aws.String(fmt.Sprintf("/aws-service-role/%s/", service_name)),
+		}
+
+		resp, err := conn.ListRoles(params)
+
+		if err != nil || resp == nil {
```

Comment:
This logic will currently ignore errors like `PermissionDenied`. It should return all errors except on  `isAWSErr(err, iam.ErrCodeNoSuchEntityException, "")`